lib/copy_gauge_offset.cu

```diff
@@ -0,0 +1,247 @@
+#include <tune_quda.h>
+#include <gauge_field_order.h>
+#include <quda_matrix.h>
+
+#include <instantiate.h>
+
+namespace quda
+{
+
+  using namespace gauge;
+
+  /**
+     Kernel argument struct
+   */
+  template <class Float, int nColor, class Order> struct CopyGaugeOffsetArg {
+
+    static constexpr int nDim = 4;
+
+    using real = typename mapper<Float>::type;
+    using Matrix = Matrix<complex<real>, nColor>;
+
+    Order out;
+    const Order in;
+
+    int Xin[QUDA_MAX_DIM];
+    int Xout[QUDA_MAX_DIM];
+
+    int offset[QUDA_MAX_DIM];
+
+    int volume_cb;
+
+    bool collect_disperse; // Whether we are collecting (true) or dispersing (false)
```

Comment:
Could this be an class enum with just two states? I find collect_disperse=true hard to parse as collect.

Reply:
Will do that.